commit 614237996af4aada9ef46bf7918ec7a95b84e687 (refs/changes/15/1515/1)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2017-02-15T19:32:38+00:00 (2 years, 8 months ago)
    
    CNAPI-693 CNAPI crashes when Moray returns an initial empty set of servers for allocation

diff --git a/lib/common.js b/lib/common.js
index b4135a2..5b941b8 100644
--- a/lib/common.js
+++ b/lib/common.js
@@ -131,7 +131,7 @@ function getAdminIp(sysinfo) {
 
 function randSort(array) {
     /* Durstenfeld shuffle */
-    for (var i = array.length - 1; i !== 0; i--) {
+    for (var i = array.length - 1; i > 0; i--) {
         var j = Math.floor(Math.random() * (i + 1));
 
         var tmp = array[i];
